libs/librepcb/project/boards/items/bi_netline.cpp

```diff
@@ -180,6 +180,16 @@ BI_NetLineAnchor* BI_NetLine::getOtherPoint(
   }
 }
 
+Point BI_NetLine::getClosestPoint(const Point& pos) const noexcept {
+  Point origin = mStartPoint->getPosition();
+  Point direction = mEndPoint->getPosition() - origin;
+  Point lhs = pos - origin;
+  Length dotP = Point::dotProduct(lhs, direction);
+  qreal factor = dotP.toPx() / Point::dotProduct(direction, direction).toPx();
+  Point result = origin + direction * factor;
+  return result;
```

Comment:
Calculating the nearest point on a line is already implemented in `Toolbox::nearestPointOnLine()`. Could you use that method instead of creating a new, redundant way to do the same?

Reply:
Thanks for the heads-up. I will replace my implementation. That makes also the int and qreal implementations obsolete.